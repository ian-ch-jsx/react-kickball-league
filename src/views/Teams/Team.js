import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeamById } from '../../services/teams';
import './Teams.css';

export default function Team(props) {
  const [team, setTeam] = useState([]);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      const data = await getTeamById(props.match.params.id);
      setTeam(data[0]);
      setLoading(false);
    };
    fetchData();
  }, [props.match.params.id]);
  if (loading) return <h1>loading</h1>;

  return (
    <div className="container">
      <h1>{team.name}</h1>
      <h2>Players</h2>
      {team.players.map((player) => (
        <p key={player.id}>
          <Link to={`/players/${player.id}`} key={player.id}>
            {player.name}
          </Link>
        </p>
      ))}
    </div>
  );
}
