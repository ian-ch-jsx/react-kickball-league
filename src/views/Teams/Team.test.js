import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Team from './Team';

test('renders team name', () => {
  const { container } = render(
    <MemoryRouter>
      <Team match={{ params: { id: 1 } }} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
