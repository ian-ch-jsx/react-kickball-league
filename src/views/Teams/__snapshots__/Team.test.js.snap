// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team name and players 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Bridge City Sneakers
    </h1>
    <h2>
      Players
    </h2>
    <p>
      <a
        href="/players/2"
      >
        Hans Olo
      </a>
    </p>
    <p>
      <a
        href="/players/3"
      >
        Ron Long
      </a>
    </p>
    <p>
      <a
        href="/players/4"
      >
        Shorty Biggum
      </a>
    </p>
    <p>
      <a
        href="/players/1"
      >
        Bennie Jetts
      </a>
    </p>
  </div>
</div>
`;
